File: vol1/ch1/LeetCode283-2.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        n,k=len(nums),0
        for i in range(0,n):
            if nums[i]==0: k+=1            #nums[i]为要删除的元素
            else: nums[i-k]=nums[i]         #nums[i]为要保留的元素
        for i in range(n-k,n):
            nums[i]=0

File: vol1/ch1/LeetCode283-2.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int n=nums.size();
        int k=0;
        for (int i=0;i<n;i++) {
            if (nums[i]==0) k++;            //nums[i]为要删除的元素
            else nums[i-k]=nums[i];         //nums[i]为要保留的元素
        }
        for(int i=n-k;i<n;i++)
            nums[i]=0;
  }
};

